torch/csrc/jit/passes/bailout_graph.cpp

```diff
@@ -117,7 +141,35 @@ struct BailOutGraphBuilderForNode {
     buildBailOutBlockFrom(outer_node->next());
   }
 
+  void mapLoopCounts(Node *n) {
+
+    auto loopCounts = collectLoopCounts(n);
+    for (auto lc : loopCounts) {
+      getOrAddInputForValue(lc);
+    }
+  }
+
   std::shared_ptr<Graph> buildBailOutGraphFrom(Node* n) {
+
+    // BailOut graphs are built lazily when a bailout is triggered
```

Comment:
This scheme seems fragile. What two pieces of code need to be kept in sync? What features of that code ensure it will stay true? It seems like loop counters are the one exception, but where is the code that causes that? 

It feels like it would be more robust to accept the order of the live captured variables as the true order of the inputs and to build the bailout graph to respect that order.